apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -633,6 +633,55 @@ static void tc_pthread_pthread_create_exit_join(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+* @fn                   :tc_pthread_pthread_tryjoin_np
+* @brief                :test the pthread_tryjoin_np function
+* @Scenario             :1. creates a new thread with infinite loop
+*                        2. call pthread_tryjoin_np
+*                        3. cancel thread
+*                        4. call pthread_tryjoin_np again
+* API's covered         :pthread_create, pthread_cancel, pthread_tryjoin_np
+* Preconditions         :none
+* Postconditions        :none
+* @return               :void
+*/
+static void tc_pthread_pthread_tryjoin_np(void)
+{
+	int ret_chk;
+	pthread_t pid;
+	void *pexit_value = 0;
+
+	ret_chk = pthread_create(&pid, NULL, setgetname_thread, NULL);
+	TC_ASSERT_EQ("pthread create", ret_chk, OK);
+
+	/* To make sure thread is running */
+	sleep(SEC_1);
+
+	ret_chk = pthread_tryjoin_np(pid, NULL);
+	TC_ASSERT_EQ("pthread_tryjoin_np", ret_chk, EBUSY);
+
+	ret_chk = pthread_cancel(pid);
+	TC_ASSERT_EQ("pthread_cancel", ret_chk, OK);
+
+	/* To make sure thread is terminated */
+	sleep(SEC_1);
+
+	ret_chk = pthread_tryjoin_np(pid, &pexit_value);
+	TC_ASSERT_EQ("pthread_tryjoin_np", ret_chk, ESRCH);
+
+	ret_chk = pthread_create(&pid, NULL, do_nothing_thread, NULL);
```

Comment:
do_nothing_thread is removed in your another commit. Please check this.

Reply:
I added this at 2nd commit and I removed do_nothing_thread at 3rd commit. I refacored it at 3rd commit with 2nd commit added line together. So it is right. At 2nd commit, another thread is not perfectly matched to my usages.